### DISCUSSION TITLE
Native UI with Dart-native

### DISCUSSION THREAD

**Discussion by squirelboy360** [dart-native/dart_native#119] [Ideas]
I find dart-native very interesting and would like making mention that I started making a dart framework for dart to native UI to diversify the UI building approach in dart instead of just flutter's abstracted UI approach . I came across this dart package and thought i could suggest this to the team before jumping in to things by myself since you seemed to have made alot of things that could be reutilised in this quest. With all these said, it is just a suggestion as it seems frameworks like jaspr are now diversifying the use of dart for native dom (web) too. We are yet to see dart to native mobile ui bindings and i hope the team of dart-native considers this.  Thanks

**Comment by yulingtianxia** [comment]
@squirelboy360 This is quite an innovative idea, however, I am concerned that it might compromise the self-rendering architecture of the Flutter framework and potentially result in performance challenges. It would be advisable to consider bridging to native UI components only when absolutely essential. I am keen to learn more about the primary objectives and anticipated advantages of this strategy, as well as the reasons for not opting for a framework akin to React Native.

**Reply by squirelboy360** [reply]
@yulingtianxia You are actually right about this. I am currently using react native and it is cool. But tbh I love the dart syntax and it's about time dart diversified. Now we have native dom frameworks in dart, why not apps with all the native primitive bindings exposed to dart. Flutter canvas rendering method is great but not for everyone as if you realise, flutter is always trying to catch up with native instead of being native itself. And now it seems technology can now allows as to communicate with native ui layer and dart side by sharing memory. Also a quick mention here that this is not flutter.  But another framework built from ground up that does not leverage the flutter engine as I believe binding native ui to dart should not be as tough as building your own engine like chrome or impeller. The only issue is creating thousands of bindings which can be solved by proving to the community the potency of this new framework like the ability to make a simple todo list app. Then with some easy docs people can start contributing. 
Again react native is great but jus like .net Maui that also uses native bindings but in c++ we need one for dart. Cause I can assure you new flutter devs enter flutter thinking its native then pivot along the line to react like I did years back. But missed the dart language.

**Reply by squirelboy360** [reply]
I actually started something small to validate if this could work. And I haven't been able to die to building several projects at once. I would return to it later to validate it bit for now I wanted to spread some awareness to see what you might have to say. Considering how interesting your package was to me and how you masterminded it. https://github.com/squirelboy360/frameworks/tree/main/direct_native

**Reply by squirelboy360** [reply]
I decided on this syntax as that makes it easier for communication with bindings ![Screenshot_2024-10-24-10-56-44-65_320a9a695de7cdce83ed5281148d6f19.jpg](https://github.com/user-attachments/assets/14c9891c-9072-4d70-86d9-2c1b74b313fe)
